### ISSUE TITLE
Don't set onChange validation to "valid" for field with async rule with no sync rules

### ISSUE THREAD

**Issue by kettanaito** [kettanaito/react-advanced-form#152]
## What
I suggest we don't mark field as `valid` if it has `asyncRule`, but doesn't have any sync rules applied.

## Why
When you edit async field, it gets marked as `valid`, and then validation happens on blur. This is confusing in case there is no sync validation to indicate as `valid`.

## How
1. Check if field has any sync rules during the validation.
1. If it's async, and has no sync rules, don't mark it as `valid` unless async validation happens.

**Comment by kettanaito** [comment]
I would also appreciate your thoughts on this @Vidlec. Thanks!

**Comment by kettanaito** [comment]
The root of the issue lies in the `fieldUtils.shouldValidate()` method.

Basically, it resolves whenever the validation is needed, regardless of which validation *type* was requested to be checked.

```
onChange > SyncValidation > shouldValidate? Sure, has "asyncRule" > Requests new validity state
```

**Comment by kettanaito** [comment]
## Changelog
* `fieldUtils.shouldValidate()` was removed
* Its implementation was modified and added as `ValidationType.shouldValidate()`

## TODO
- [ ] Write test(s)
- [ ] `asyncRule` Promise resolving with timeout throws the following error:
```
Uncaught (in promise) TypeError: resolved.getIn is not a function
    at Sequence.sequenceIterator [as iterator] (http://localhost:6006/static/preview.bundle.js:11701:27)
    at Sequence._callee$ (http://localhost:6006/static/preview.bundle.js:58391:28)
    at tryCatch (http://localhost:6006/static/preview.bundle.js:27681:40)
    at Generator.invoke [as _invoke] (http://localhost:6006/static/preview.bundle.js:27915:22)
    at Generator.prototype.(anonymous function) [as next] (http://localhost:6006/static/preview.bundle.js:27733:21)
    at step (http://localhost:6006/static/preview.bundle.js:58294:191)
    at http://localhost:6006/static/preview.bundle.js:58294:361
    at <anonymous>
```

**Comment by kettanaito** [comment]
The issue is covered and published under [`0.18.3`](https://github.com/kettanaito/react-advanced-form/commit/852fc6ee4aa0f6ede46fb36cc91f8382cffcc6d0). Closing.
